src/frontend/templates/pages/game_network_groups.html

```diff
@@ -0,0 +1,49 @@
+{% extends 'base.html' %}
+
+{% block title %}
+{% if not run %}
+Games
+{% else %}
+Games for {{run.name}}
+{% endif %}
+{% endblock %}
+
+{% block content %}
+
+{% if not run %}
+No currently active run to view games for.
+{% elif viewing_current_run %}
+<h1 class="title">Games for Current Run ({{run.name}}) by Network</h1>
+{% else %}
+<h1 class="title">Games for {{run.name}} by Network</h1>
+{% endif %}
+
+{% if run and object_list|length <= 0 %}
+No games found for this run!
+{% elif run and object_list|length > 0 %}
+<table>
+  <tr>
+    <th> Network </th>
+    <th>  </th>
+    <th>  </th>
+  </tr>
+  {% for network in object_list %}
+  <tr>
+    <td> {{ network.name }} </td>
+    <td> <a href="{% url 'training_games_list' run.name network.name 1 %}"> Training Games </td>
+    <td> <a href="{% url 'rating_games_list' run.name network.name 1 %}"> Rating Games </td>
+  </tr>
+  {% endfor %}
```

Comment:
Do you mean "inefficient" for performance, or "inefficient" for the user UI?

If you mean "inefficient" for the user UI, then I'm curious what a better design is. The design you suggests has a major problem which is that the number of games is so extremely large that just having the whole thing in a list view is not very useful for the user. The user needs a way to filter the games by different criteria, and jump to different sections of the games.

I didn't know how to implement a general filtering mechanism, so I figured that grouping by networks would be a reasonable choice. A lot of possible applications would be handled by grouping by networks (e.g. the user wants to see all the games that contribute to a network's Elo, or the user wants to compare how two different networks play). Also, since network ordering loosely corresponds to chronological ordering (not completely, but very close), this means that applications that want to get recent games, or games from a certain time range, can also do so approximately.

But if you think we can have a single list page and implement some effective and much more general way of user filtering, then that would be better.

Reply:
> I didn't know how to implement a general filtering mechanism, so I figured that grouping by networks would be a reasonable choice. A lot of possible applications would be handled by grouping by networks (e.g. the user wants to see all the games that contribute to a network's Elo, or the user wants to compare how two different networks play). Also, since network ordering loosely corresponds to chronological ordering (not completely, but very close), this means that applications that want to get recent games, or games from a certain time range, can also do so approximately.

Indeed. We can use django filter later https://django-filter.readthedocs.io/en/master/guide/usage.html#the-view

---

On think use need is the latest games he contributed, but it will be in contribution page.